reference/functions/pgroonga-condition.md

```diff
@@ -84,23 +84,228 @@ Please refer to [Calling Functions][sql-syntax-calling-funcs] for information ab
+   Filter: (name &^ '(_p_G,,,,pgroonga_tag_name_index,)'::pgroonga_condition)
+   Rows Removed by Filter: 2
+ Planning Time: 0.871 ms
+ Execution Time: 2.352 ms
+(5 rows)
+
+SELECT *
+  FROM tags
+ WHERE name &^ pgroonga_condition('_p_G',
+                                  index_name => 'pgroonga_tag_name_index');
+   name
+-----------
+ PGroonga
+ pglogical
+(2 rows)
+```
+
+`index_name`を指定しない場合（つまり、`NormalizerNFKC150("remove_symbol", true)`が参照できない場合）は、次のように「`PGroonga`」と「`pglogical`」はヒットしません。
+
+```sql
+EXPLAIN ANALYZE
+SELECT *
+  FROM tags
+ WHERE name &^ pgroonga_condition('_p_G');
+                                            QUERY PLAN
+--------------------------------------------------------------------------------------------------
+ Seq Scan on tags  (cost=0.00..1043.60 rows=1 width=32) (actual time=0.032..0.032 rows=0 loops=1)
+   Filter: (name &^ '(_p_G,,,,,)'::pgroonga_condition)
+   Rows Removed by Filter: 4
+ Planning Time: 0.910 ms
+ Execution Time: 0.053 ms
+(5 rows)
+
+SELECT *
+  FROM tags
+ WHERE name &^ pgroonga_condition('_p_G');
+
+ name
+------
+(0 rows)
+```
+
+このように、`index_name`を指定することで、シーケンシャルサーチ実行時でもインデックスサーチ実行時でも検索結果が変わらないようにできます。
+
+### Specify `weights`
+
+カラム毎に異なるweight（重要度）を設定する方法を紹介します。
+これにより、「タイトルを本文よりも重要視する」を実現できます。
+
+`pgroonga_condition('keyword', ARRAY[weight1, weight2, ...])` を使います。
+`weight1`、 `weight2`でカラム毎の重要度を指定します。
+
+Here are sample schema and data:
+
+```sql
+DROP TABLE IF EXISTS memos;
+CREATE TABLE memos (
+  title text,
+  content text
+);
+
+CREATE INDEX pgroonga_memos_index
+    ON memos
+ USING pgroonga ((ARRAY[title, content]));
+
+INSERT INTO memos VALUES ('PostgreSQL', 'PostgreSQLはリレーショナル・データベース管理システムです。');
+INSERT INTO memos VALUES ('Groonga', 'Groongaは日本語対応の高速な全文検索エンジンです。');
+INSERT INTO memos VALUES ('PGroonga', 'PGroongaはインデックスとしてGroongaを使うためのPostgreSQLの拡張機能です。');
+INSERT INTO memos VALUES ('コマンドライン', 'groongaコマンドがあります。');
+```
+
+指定したクエリーによりマッチしたレコードを探すためには[`pgroonga_score function`][pgroonga-score-function]を使えます。
+
+```sql
+SELECT *, pgroonga_score(tableoid, ctid) AS score
+  FROM memos
+ WHERE ARRAY[title, content] &@~
+       pgroonga_condition('Groonga OR PostgreSQL', ARRAY[5, 1])
+ ORDER BY score DESC;
+      title      |                                  content                                  | score 
+----------------+---------------------------------------------------------------------------+-------
+ Groonga        | Groongaは日本語対応の高速な全文検索エンジンです。                         |     6
+ PostgreSQL     | PostgreSQLはリレーショナル・データベース管理システムです。                |     6
+ PGroonga       | PGroongaはインデックスとしてGroongaを使うためのPostgreSQLの拡張機能です。 |     2
+ コマンドライン | groongaコマンドがあります。                                               |     1
+(4 rows)
+```
+
+上の例では、`ARRAY[title, content] &@~ pgroonga_condition('Groonga OR PostgreSQL', ARRAY[5, 1])`と指定しているので、タイトルが本文より5倍重要としています。
+`title`カラムに「`Groonga`」または「`PostgreSQL`」があるレコードの方が`content`カラムに「`Groonga`」または「`PostgreSQL`」がある方よりスコアーが高いことを確認できます。
```

Comment:
I
```suggestion
In above example, the title is 5 times weighted that the main text because `ARRAY[title, content] &@~ pgroonga_condition('Groonga OR PostgreSQL', ARRAY[5, 1])` is specified.
You can see the score is higher for the records with`Groonga` or `PostgreSQL` in the `title` column compare to the records with `Groonga` or `PostgreSQL`  in the `content` column.
```

Reply:
```suggestion
In above example, the title is 5 times weighted that the main text because `ARRAY[title, content] &@~ pgroonga_condition('Groonga OR PostgreSQL', ARRAY[5, 1])` is specified.
You can see the score is higher for the records with`Groonga` or `PostgreSQL` in the `title` column compared to records with `Groonga` or `PostgreSQL` in the `content` column.
```